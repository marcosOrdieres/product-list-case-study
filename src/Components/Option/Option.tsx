import React from 'react';
import styled from "styled-components";

export const StyledOption = styled.option`
  color: ${(props) => (props.selected ? "lightgrey" : "black")};
`;

export interface OptionProps {
  selected?: boolean,
  value?: string,
};
const Option = (props: OptionProps) => (
  <StyledOption selected={props.selected}>
    {props.value}
  </StyledOption>
);

export default Option;