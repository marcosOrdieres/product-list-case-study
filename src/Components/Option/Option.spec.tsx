import React from 'react';
import { render } from '@testing-library/react';

import Option from './Option';

describe('Option', () => {
  it('should render Option and match styles', () => {
    const { asFragment } = render(
      <Option selected={true} value='' />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});