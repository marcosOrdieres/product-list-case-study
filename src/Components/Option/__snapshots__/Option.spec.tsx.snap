// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option should render Option and match styles 1`] = `
<DocumentFragment>
  <option
    class="sc-bdfBQB fSBjyw"
  />
</DocumentFragment>
`;
